pkg/controller/kubelet-config/kubelet_config_controller.go

```diff
@@ -632,6 +632,21 @@ func (ctrl *Controller) syncKubeletConfig(key string) error {
 			updateOriginalKubeConfigwithNodeConfig(nodeConfig, originalKubeConfig)
 		}
 
+		if !isNotFound {
+			// When a Machine Config already exists, let's check if it needs an update on the feature Gates
+			match, err := machineConfigFeatureGatesMatchesOriginalFeatureGates(mc, originalKubeConfig)
+			if err != nil {
+				return ctrl.syncStatusOnly(cfg, err, "could not check if original and machine config feature gates match: %v", err)
+			}
+			if match && cfg.Status.ObservedGeneration >= cfg.Generation && cfg.Status.Conditions[len(cfg.Status.Conditions)-1].Type == mcfgv1.KubeletConfigSuccess {
```

Comment:
We need to check the length of Conditions here just in case we index to 0-1 (-1), so it doesn't crash.